polaris-react/src/components/Button/Button.module.scss

```diff
@@ -316,6 +316,13 @@
   --pc-button-icon-fill_disabled: var(--p-color-icon-disabled);
 }
 
+.Button.iconOnly:is(.variantMonochromePlain) {
```

Comment:
Could we reduce the `.Button` part of the selector to lower the specificity and match the other selectors?

```suggestion
.iconOnly:is(.variantMonochromePlain) {
```